tests/components/create_comment/create_comment.test.jsx

```diff
@@ -0,0 +1,99 @@
+    beforeEach(() => {
+        window.mm_config = {};
+        window.mm_config.EnableEmojiPicker = 'true';
+    });
+
+    const channelId = 'g6139tbospd18cmxroesdk3kkc';
+    const rootId = '';
+    const latestPostId = '3498nv24823948v23m4nv34';
+
+    test('should match snapshot, empty comment', () => {
+        const wrapper = shallow(
+            <CreateComment
+                channelId={channelId}
+                rootId={rootId}
+                draft={{
+                    message: '',
+                    uploadsInProgress: [],
+                    fileInfos: []
+                }}
+                enableAddButton={false}
+                ctrlSend={true}
+                latestPostId={latestPostId}
+                getSidebarBody={jest.fn()}
+                onUpdateCommentDraft={jest.fn()}
+                onSubmit={jest.fn()}
+                onResetHistoryIndex={jest.fn()}
+                onMoveHistoryIndexBack={jest.fn()}
+                onMoveHistoryIndexForward={jest.fn()}
+                onEditLatestPost={jest.fn()}
+            />
+        );
+
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('should match snapshot, comment with message', () => {
+        const wrapper = shallow(
+            <CreateComment
+                channelId={channelId}
+                rootId={rootId}
+                draft={{
+                    message: 'Test message',
+                    uploadsInProgress: [],
+                    fileInfos: []
+                }}
+                enableAddButton={false}
+                ctrlSend={true}
+                latestPostId={latestPostId}
+                getSidebarBody={jest.fn()}
+                onUpdateCommentDraft={jest.fn()}
+                onSubmit={jest.fn()}
+                onResetHistoryIndex={jest.fn()}
+                onMoveHistoryIndexBack={jest.fn()}
+                onMoveHistoryIndexForward={jest.fn()}
+                onEditLatestPost={jest.fn()}
+            />
+        );
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('should match snapshot, non-empty message and uploadsInProgress + fileInfos', () => {
+        const wrapper = shallow(
+            <CreateComment
+                channelId={channelId}
+                rootId={rootId}
+                draft={{
+                    message: 'Test message',
+                    uploadsInProgress: [{}],
+                    fileInfos: [{}, {}, {}]
+                }}
+                enableAddButton={false}
+                ctrlSend={true}
+                latestPostId={latestPostId}
+                getSidebarBody={jest.fn()}
+                onUpdateCommentDraft={jest.fn()}
+                onSubmit={jest.fn()}
+                onResetHistoryIndex={jest.fn()}
+                onMoveHistoryIndexBack={jest.fn()}
+                onMoveHistoryIndexForward={jest.fn()}
+                onEditLatestPost={jest.fn()}
+            />
+        );
+        expect(wrapper).toMatchSnapshot();
+    });
+});
```

Comment:
Test for CreateComment is low at ~20%. Could you please add more tests, at least for the following?
- toggleEmojiPicker (assert state change)
- hideEmojiPicker (assert state change)
- handlePostError (assert state change)
- handleUploadError (assert state change)
- getFileCount
- showPostDeletedModal (assert state change)
- hidePostDeletedModal (assert state change)

![screen shot 2017-11-09 at 3 15 21 am](https://user-images.githubusercontent.com/5334504/32569514-49042592-c4fc-11e7-82a3-a5fc3487e683.png)

Reply:
Yup, no problem

Reply:
@saturninoabril added more tests 👍